apis/python/src/tiledbsoma/handles.py

```diff
@@ -0,0 +1,239 @@
+"""Abstractions to more easily manage read and write access to TileDB data.
```

Comment:
Filename- how about smth like `tiledb_handle_wrappers.py`?

Reply:
I went with `tdb_handles.py` in the interest of keeping it short and reducing repetition.